docs/milestone-2.md

```diff
@@ -0,0 +1,297 @@
+# Milestone 2 - security, authentication, & reputation
+
+This second milestone adds all the security features necessary for production use of the software including relay and validator authentication, merkle proof of payment, and payload validity fraud proof.
+
+## Specification
+
+### Authentication
+
+```sequence
+participant mev_boost
+participant consensus
+participant relays
+Title: Authentication
+mev_boost->consensus: consensus_getFeeRecipientsV1
+consensus-->mev_boost: consensus_getFeeRecipientsV1 response
+mev_boost->relays: relay_publishFeeRecipientsV1
+mev_boost->relays: relay_getValidatorStatusV1
+```
+
+1. on startup, mev-boost requests a list of [`AuthenticatedFeeRecipient`](#authenticatedfeerecipient) from the consensus client using a new [`consensus_getFeeRecipientsV1`](#consensus_getfeerecipientsv1) method.
```

Comment:
IMO this should be an mev-boost endpoint which consensus clients must `POST` to, something like `builder_updateFeeRecipientsV1`. I don't think this API should be made bidirectional if possible. Other endpoints in the builder API should respond with a specific status code if it doesn't yet have a suggested fee recipient.  If a consensus client ever receives a payload header from mev-boost with an incorrect `suggested_fee_recipient`, they can update it with another call to `builder_updateFeeRecipientsV1`